src/System.Private.CoreLib/shared/System/Memory.cs

```diff
@@ -258,6 +258,35 @@ public Memory<T> Slice(int start, int length)
             return new Memory<T>(_object, _index + start, length);
         }
 
+        /// <summary>
+        /// Forms a slice out of the given memory, beginning at 'startIndex'
+        /// </summary>
+        /// <param name="startIndex">The index at which to begin this slice.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Memory<T> Slice(Index startIndex)
```

Comment:
Do we need to append the type name to the parameter name? An `Index` is definitely an index, rather than a length or anything else. Even the `int` parameter is named without suffix.

Reply:
Index in the name is not really the type name more than just it is the start index to slice. I agree this is not consistent with the method that takes int parameter but I am not seeing this a serious issue to fix and even it is too late to change it now.